go-apps/meep-wais/server/wais.go

```diff
@@ -0,0 +1,802 @@
+/*
+ * Copyright (c) 2020  InterDigital Communications, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	sbi "github.com/InterDigitalInc/AdvantEDGE/go-apps/meep-wais/sbi"
+	dkm "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-data-key-mgr"
+	httpLog "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-http-logger"
+	log "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-logger"
+	redis "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-redis"
+	sm "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-sessions"
+	clientNotif "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-wais-notification-client"
+
+	"github.com/gorilla/mux"
+)
+
+const moduleName = "meep-wais"
+const waisBasePath = "/wai/v2/"
+const waisKey string = "wais:"
+const logModuleWAIS string = "meep-wais"
+
+var redisAddr string = "meep-redis-master.default.svc.cluster.local:6379"
+var influxAddr string = "http://meep-influxdb.default.svc.cluster.local:8086"
+var postgisHost string = "meep-postgis.default.svc.cluster.local"
+var postgisPort string = "5432"
+
+const assocStaSubscriptionType = "AssocStaSubscription"
+
+// const staDataRateSubscriptionType = "StaDataRateSubscription" //no used at the moment
+
+var assocStaSubscriptionMap = map[int]*Subscription{}
+var subscriptionExpiryMap = map[int][]int{}
+var currentStoreName = ""
+
+var WAIS_DB = 5
```

Comment:
Is there a reason to keep this in DB index 5? Same question applies to RNIS.
Now that data key structure is in place to clearly identify what data belongs to which service, we could use a single DB (0).

Reply:
i agree with you... i admit i didn't know what to do with those.. so i kept it as 5. Actually, either all services use the same DB index, or the whole project uses the same DB index.. i guess we need to conclude on that...